workflows/pipe-common/scripts/autoscale_sge.py

```diff
@@ -963,17 +1038,25 @@ def scale(self):
                 if now >= self.latest_running_job.datetime + self.scale_down_timeout:
                     Logger.info('Latest job started more than %s seconds ago. Scaling down is required.' %
                                 self.scale_down_timeout.seconds)
-                    self._scale_down(running_jobs, additional_hosts)
+                    self._scale_down(running_jobs, additional_hosts, now)
                 else:
                     Logger.info('Latest job started less than %s seconds. '
                                 'Scaling down is not required.' % self.scale_down_timeout.seconds)
             else:
                 Logger.info('There are no previously running jobs. Scaling down is required.')
-                self._scale_down(running_jobs, additional_hosts)
+                self._scale_down(running_jobs, additional_hosts, now)
         Logger.info('Finish scaling step at %s.' % self.clock.now())
         post_scale_additional_hosts = self.host_storage.load_hosts()
         Logger.info('There are %s additional pipelines.' % len(post_scale_additional_hosts))
 
+    @staticmethod
+    def update_running_jobs_host_activity(running_jobs, host_storage, scaling_step_start):
```

Comment:
It feels like this method should be a part of `host_storage` implementations.

Reply:
Done